### PR TITLE
Move files_external to IBootstrap and remove some deprecated stuff in the process

### PR DESCRIPTION
**Pull request by icewind1991**

### CODE REVIEW

**Review by blizzz — APPROVED** [review]
👍 – sharepoint master also continues to work with this

File: apps/files_external/lib/Command/Create.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ protected function configure() {
 	}
 
 	protected function execute(InputInterface $input, OutputInterface $output): int {
-		$user = $input->getOption('user');
+		$user = (string) $input->getOption('user');
```
Comment (MorrisJobke) [inline apps/files_external/lib/Command/Create.php:123]: The return value could be `string|string[]|bool|null`. MInd to remove this change for now and then we look into make this correct?

**Review by ChristophWurst — APPROVED** [review]
:+1: :elephant:

File: apps/files_external/lib/Command/Create.php (hunk 1 above)
Comment (icewind1991) [reply apps/files_external/lib/Command/Create.php:123]: psalm wants the cast

Comment (MorrisJobke) [reply apps/files_external/lib/Command/Create.php:123]: But casting and array to string will cause issues, no?

**Comment by blizzz** [comment]
rebased

**Pull request merged**